gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -214,18 +215,31 @@ public void onAddSpec(Spec addedSpec) {
           scheduleJob(jobConfig, null);
           if (PropertiesUtils.getPropAsBoolean(jobConfig, ConfigurationKeys.FLOW_RUN_IMMEDIATELY, "false")) {
             _log.info("RunImmediately requested, hence executing FlowSpec: " + addedSpec);
-            this.jobExecutor.execute(new NonScheduledJobRunner(jobConfig, null));
+            scheduleJobImmediately(jobConfig, null);
```

Comment:
Why add this instead of a delete flag to `NonScheduledJobRunner`?

Reply:
My understanding was scheduleJobImmediately was to handle scheduled case and NonScheduledJobRunner was to handle non-scheduled case. There is no clear differentiation between scheduleJobImmediately vs. NonScheduledJobRunner. If it caused you a confusion, I will remove it.